src/sagemaker/session.py

```diff
@@ -2340,13 +2342,17 @@ def _map_training_config(
             training_job_definition["VpcConfig"] = vpc_config
 
         if enable_network_isolation:
-            training_job_definition["EnableNetworkIsolation"] = True
+            training_job_definition["EnableNetworkIsolation"] = enable_network_isolation
```

Comment:
nit: Isnt this redundant, if F will never come here?

Reply:
This was caught from a customer reported issue.
`enable_network_isolation` can be a Pipeline variable like ParameterBoolean, which is not parsed in compile time (SDK stage). 
However, in such case, it will go into this line and hard code a True to it, which means the ParameterBoolean object will be lost in the Pipeline definition and accordingly users are not able to update the this parameter in runtime.